opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,27 +12,161 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# type: ignore
+
 from sys import version_info
+from typing import Dict, Tuple, Union
+
+# This is a cache to avoid going through creating a dictionary for all entry
+# points for 3.7 every time the entry points function is called.
+result_entry_points_37 = {}
 
 # FIXME remove this when support for 3.7 is dropped.
 if version_info.minor == 7:
     # pylint: disable=import-error
-    from importlib_metadata import entry_points, version  # type: ignore
+    from importlib_metadata import EntryPoint
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import version
+
+    def entry_points(
+        group: str = None, name: str = None
+    ) -> Union[Tuple[EntryPoint], Dict[str, Tuple[EntryPoint]]]:
```

Comment:
Can you add `@typing.overload`s to describe when this function returns a Tuple vs a Dict? IIUC dict is only returned when group and name are None

Reply:
```suggestion
    ) -> Union[Tuple[EntryPoint, ...], Dict[str, Tuple[EntryPoint, ...]]]:
```

also for variable length tuple

Reply:
Fixed